backend/routers/questions.router.test.js

```diff
@@ -0,0 +1,115 @@
+// Mock and import Question model
+jest.mock('../models/question.model');
+const { Question } = require('../models');
+
+// Import question router
+const questionsRouter = require('./questions.router');
+
+// Create a test app with Express
+const express = require('express');
+const supertest = require('supertest');
+const testapp = express();
+// Allow for body parsing of JSON data
+testapp.use(express.json());
+// Allow for body parsing of HTML data
+testapp.use(express.urlencoded({ extended: false }));
+testapp.use('/api/questions/', questionsRouter);
+const request = supertest(testapp);
+
+describe('Unit tests for questions router', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('READ', () => {
+    // Mock question data
+    const mockQuestions = [
+      {
+        id: 1,
+        questionText: 'What is your favorite color?',
+        htmlName: 'favoriteColor',
+        answers: {
+          answerOneText: 'Red',
+          answerTwoText: 'Blue',
+          answerThreeText: 'Green',
+          answerFourText: 'Yellow',
+        },
+      },
+      {
+        id: 2,
+        questionText: 'What is your favorite food?',
+        htmlName: 'favoriteFood',
+        answers: {
+          answerOneText: 'Pizza',
+          answerTwoText: 'Cheeseburger',
+          answerThreeText: 'Sushi',
+          answerFourText: 'Chicken',
+        },
+      },
+    ];
+
+    it('should return all questions with GET /api/questions', async (done) => {
```

Comment:
Because the logic for responding to requests in present in this router, we should test for fail cases. After this test for a successful `Question.find()`, we should add something like this:

```
    it('should return 400 status code when there is an error with GET /api/questions', async (done) => {
      // Mock the Question.find() method
      Question.find.mockRejectedValue(new Error('Database error'));

      // Mock the request to the API
      const response = await request.get('/api/questions');

      // Tests
      expect(Question.find).toHaveBeenCalled();
      expect(response.status).toBe(400);

      // Marks completion of tests
      done();
    });
    ```